src/host/exe/exemain.cpp

```diff
@@ -175,35 +221,64 @@ int CALLBACK wWinMain(
     HRESULT hr = args.ParseCommandline();
     if (SUCCEEDED(hr))
     {
-        if (ShouldUseLegacyConhost(args))
+        if (args.ShouldRunAsComServer())
         {
-            if (args.ShouldCreateServerHandle())
+            try
             {
-                hr = E_INVALIDARG;
-            }
-            else
-            {
-                hr = ValidateServerHandle(args.GetServerHandle());
+                // OK we have to do this here and not in another method because
+                // we would either have to store the module ref above in some accessible
+                // variable (which would be awful because of the gigantic template name)
+                // or we would have to come up with some creativity to extract it out
+                // of the singleton module base without accidentally having WRL
+                // think we're recreating it (and then assert because it's already created.)
+                //
+                // Also this is all a problem because the decrementing count of used objects
+                // in this module in WRL::Module base doesn't null check the release notifier
+                // callback function in the OutOfProc variant in the 18362 SDK. So if anything
+                // else uses WRL directly or indirectly, it'll crash if the refcount
+                // ever hits 0.
+                // It does in the 19041 SDK so this can be cleaned into its own class if

```

Comment:
file a todo so we don't forget?

Reply:
I don't really think it needs to move into its own class unless someone gets a refactoring bug one day. I don't think scheduling future work for it is necessary.